cobra/manipulation/delete.py

```diff
@@ -39,20 +43,23 @@ def prune_unused_reactions(cobra_model):
     Parameters
     ----------
     cobra_model: cobra.Model
-        the model to remove unused reactions from
+    the model to remove unused reactions from
 
     Returns
     -------
+    model
+    input model with unused reactions removed
     list
-        list of reactions that were removed
+    list of reactions that were removed
     """
     pruned_reactions = []
-    reactions_to_prune = [x for x in cobra_model.reactions
+    output_model = cobra_model.copy()
+    reactions_to_prune = [x for x in output_model.reactions
                           if len(x._metabolites) == 0]
     for the_reaction in reactions_to_prune:
```

Comment:
This loop doesn't make any sense any longer. Remove it and simply change your line to
```python
output_model.remove_reactions(reactions_to_prune)
```